engine/crates/engine-v2/codegen/src/loader.rs

```diff
@@ -226,7 +226,7 @@ fn parse_union_kind(name: &str, directives: Iter<'_, Directive<'_>>) -> domain::
         }
     } else {
         domain::UnionKind::Record(domain::RecordUnion {
-            indexed: parse_indexed(name, directives),
+            indexed: parse_indexed(name, directives.clone()),
```

Comment:
Do we need this clone?

Reply:
As in do we want to add a clone here specifically for testing? It's probably a non issue but thought I'd ask

Reply:
We're not adding a clone specifically for testing.  We are adding a clone because I updated `cynic-parser` and `Iter` (which directives is an instance of) is no longer `Copy`.  This code will not compile if I remove the `clone`.  So yes we need this clone.

It is incredibly inexpensive, and this is also just codegen code that is not performance critical.  

tldr: yes

Reply:
I see, was wondering why it changed. Ok sounds good